README.md

```diff
@@ -79,17 +75,11 @@ fmt.Println("Published message")
 Create a Consumer:
 
 ```go
-client, err := pulsar.NewClient(pulsar.ClientOptions{
-    URL: "pulsar://localhost:6650",
-})
+client, err := pulsar.NewClient("pulsar://localhost:6650")
 
 defer client.Close()
 
-consumer, err := client.Subscribe(pulsar.ConsumerOptions{
-        Topic:            "my-topic",
-        SubscriptionName: "my-sub",
-        Type:             pulsar.Shared,
-    })
+consumer, err := client.Subscribe("my-sub", WithTopics("my-topic"), WithSubscriptionType(Shared))
```

Comment:
Given that topic is a mandatory argument, why is it being treated differently than subscription name?

Reply:
topic has three different form: single topic, topic list as well as topic pattern, topic pattern is different from single topic or topic list, so I need methods: WithTopics and WithTopicPattern to distinguish.

Reply:
@merlimat topic has three different form: single topic, topic list as well as topic pattern, topic pattern is different from single topic or topic list, so I need methods: WithTopics and WithTopicPattern to distinguish.